modules/caddyhttp/responsewriter.go

```diff
@@ -342,3 +363,23 @@ var (
 
 	_ io.WriterTo = (*hijackedConn)(nil)
 )
+
+// lengthReader is an io.ReadCloser that keeps track of the
+// number of bytes read from the request body.
+// This wrapper is for http request process only. If the underlying
+// conn hijacked by a websocket session. ResponseRecorder will
+// update the Length field.
+type lengthReader struct {
+	Source io.ReadCloser
+	Length int
```

Comment:
Just at first glance, is there any need for these fields to be exported? If not, could you lowercase them please?

Reply:
Sure.